gnc/sub8_trajectory_generator/src/sub8_state_space.cc

```diff
@@ -0,0 +1,34 @@
+/**
+ * Author: Patrick Emami
+ * Date: 9/21/15
+ */
+
+#include "sub8_state_space.h"
+
+ompl::base::State* sub8::trajectory_generator::Sub8StateSpace::allocState() const {
+	StateType *state = new StateType();
```

Comment:
Right, so this was still a point of confusion for me at the time. I'll just be using the angular velocity vector, I see now how using the time derivative of the Euler angles wouldn't go over very well in this situation. I'll make the relevant changes to reflect this. 



> On Sep 23, 2015, at 3:57 PM, Forrest Voight <notifications@github.com> wrote:
> 
> In gnc/sub8_trajectory_generator/src/sub8_state_space.cc:
> 
> > @@ -0,0 +1,34 @@
> > +/**
> > + * Author: Patrick Emami
> > + * Date: 9/21/15
> > + */
> > +
> > +#include "sub8_state_space.h"
> > +
> > +ompl::base::State* sub8::trajectory_generator::Sub8StateSpace::allocState() const {
> > +	StateType *state = new StateType();
> Why are you using time derivatives of Euler angles for angular velocity instead of just the angular velocity vector? From your comments, it looks like that's what you're actually doing already ("Get the angular velocity about the y axis"). At least from what I've seen, all that does is introduce more complex Jacobians everywhere and add singularities. It was only used in the old trajectory generator because it worked directly on each Euler angle separately, but AFAIK your approach won't.
> …
> —
> Reply to this email directly or view it on GitHub.
>